data_steward/cdr_cleaner/manual_cleaning_rules/dc732_suppress.py

```diff
@@ -0,0 +1,445 @@
+
+
+def main(project_id, sandbox_dataset_id, concept_lookup_table, target_dataset_ids):
+    """
+    Identify, log and remove rows with suppresed concepts
+    Note: Rows are appended to log and backup tables if they already exist
+    
+    :param project_id: Identifies the project containing all associated datasets
+    :param sandbox_dataset_id: Identifies a dataset to store log and backup records
+    :param concept_lookup_table: Identifies the table containing the concepts to suppress
+    :param target_dataset_ids: List of datasets to retract data from
+    
+    """
+    # get suppress row info
+    rows_dest_table = f'{sandbox_dataset_id}.{ROWS_RESOURCE_NAME}'
+    tables_to_suppress_df = get_tables_to_suppress_df(project_id, target_dataset_ids)
+    suppress_rows_df = get_rows_to_suppress_df(tables_to_suppress_df, concept_lookup_table)
+    suppress_rows_df.to_csv(f'{ROWS_RESOURCE_NAME}.csv')
+    suppress_rows_df.to_gbq(destination_table=rows_dest_table, if_exists='append')
+
+    # get suppress summary
+    summary_dest_table = f'{sandbox_dataset_id}.{SUMMARY_RESOURCE_NAME}'
+    suppress_summary_df = get_suppress_summary_df(suppress_rows_df)
+    suppress_summary_df.to_csv(f'{SUMMARY_RESOURCE_NAME}.csv')
+    suppress_summary_df.reset_index(inplace=True)
+    suppress_summary_df.to_gbq(destination_table=summary_dest_table, if_exists='append')
+    
+    # backup rows
+    backup_rows_to_suppress(tables_to_suppress_df, rows_dest_table)
+    
+    # delete rows
+    all_delete_queries = get_delete_queries(rows_dest_table)
+    delete_query_jobs = run_delete_queries(all_delete_queries)
+    print_jobs(delete_query_jobs)
+
+
+if __name__ == '__main__':
+    PROJECT_ID = ''
+    SANDBOX_DATASET_ID = ''
+    CONCEPT_LOOKUP_TABLE = f'{SANDBOX_DATASET_ID}.expanded_deid_concepts_20200331'
+    SUPPRESS_ROWS_DEST_TABLE = f'{SANDBOX_DATASET_ID}.{ROWS_RESOURCE_NAME}'
+    SUMMARY_DEST_TABLE = f'{SANDBOX_DATASET_ID}.{SUMMARY_RESOURCE_NAME}'
+    TARGET_DATASETS = [
+    ]
+    TABLES_TO_SUPPRESS_DF = get_tables_to_suppress_df(PROJECT_ID, TARGET_DATASETS)
+    SUPPRESS_ROWS_DF = get_rows_to_suppress_df(TABLES_TO_SUPPRESS_DF, CONCEPT_LOOKUP_TABLE)
+    SUPPRESS_ROWS_DF.to_csv(f'{ROWS_RESOURCE_NAME}.csv')
+    SUPPRESS_ROWS_DF.to_gbq(destination_table=SUPPRESS_ROWS_DEST_TABLE, if_exists='append')
+    SUPPRESS_SUMMARY_DF = get_suppress_summary_df(SUPPRESS_ROWS_DF)
+    SUPPRESS_SUMMARY_DF.to_csv(f'{SUMMARY_RESOURCE_NAME}.csv')
+    SUPPRESS_SUMMARY_DF.reset_index(inplace=True)
+    SUPPRESS_SUMMARY_DF.to_gbq(destination_table=SUMMARY_DEST_TABLE, if_exists='append')
+    
+    backup_rows_to_suppress(TABLES_TO_SUPPRESS_DF, SUPPRESS_ROWS_DEST_TABLE)
+    
+    ALL_DELETE_QUERIES = get_delete_queries(SUPPRESS_ROWS_DEST_TABLE)
+    DELETE_QUERY_JOBS = run_delete_queries(ALL_DELETE_QUERIES)
+    print_jobs(DELETE_QUERY_JOBS)
```

Comment:
opt: Most of this is covered by your `main(...)` function.  Consider replacing these rows with a call to `main(...)`?